src/burnchains/db.rs

```diff
@@ -165,23 +818,74 @@ impl<'a> BurnchainDBTransaction<'a> {
         for op in block_ops.iter() {
             let serialized_op =
                 serde_json::to_string(op).expect("Failed to serialize parsed BlockstackOp");
-            let args: &[&dyn ToSql] = &[block_hash, op.txid_ref(), &serialized_op];
+            let args: &[&dyn ToSql] = &[&block_header.block_hash, op.txid_ref(), &serialized_op];
             stmt.execute(args)?;
         }
+
+        for op in block_ops.iter() {
+            if let BlockstackOperationType::LeaderBlockCommit(ref opdata) = op {
+                let bcm = BlockCommitMetadata {
+                    burn_block_hash: block_header.block_hash.clone(),
+                    txid: opdata.txid.clone(),
+                    block_height: opdata.block_height,
+                    vtxindex: opdata.vtxindex,
+                    // NOTE: these fields are filled in by the subsequent call.
+                    affirmation_id: 0,
+                    anchor_block: None,
+                    anchor_block_descendant: None,
+                };
+                self.insert_block_commit_metadata(bcm)?;
+            }
+        }
+
+        self.update_block_descendancy(indexer, block_header, burnchain)?;
         Ok(())
     }
 
-    fn commit(self) -> Result<(), BurnchainError> {
+    pub fn commit(self) -> Result<(), BurnchainError> {
         self.sql_tx.commit().map_err(BurnchainError::from)
     }
+
+    pub fn conn(&self) -> &DBConn {
+        &self.sql_tx
+    }
+
+    pub fn get_canonical_chain_tip(&self) -> Result<BurnchainBlockHeader, BurnchainError> {
+        let qry = "SELECT * FROM burnchain_db_block_headers ORDER BY block_height DESC, block_hash ASC LIMIT 1";
+        let opt = query_row(&self.sql_tx, qry, NO_PARAMS)?;
+        Ok(opt.expect("CORRUPTION: No canonical burnchain tip"))
+    }
```

Comment:
This could use `BurnchainDB::inner_get_canonical_chain_tip`

Reply:
Yup, fixed